SharedLibs/TileMapHelper/src/model/TileMapModel.ts

```diff
@@ -352,25 +504,243 @@ export class EditableTileMapLayer extends AbstractEditableLayer {
     this._alpha = alpha;
   }
 
+  isEmpty(): boolean {
+    return this._tiles.every((row) => row.every((cell) => cell === 0));
+  }
+
+  reduceDimensions(
+    columnsToPop: number,
+    columnsToShift: number,
+    rowsToPop: number,
+    rowsToShift: number
+  ) {
+    const initialRowCount = this._tiles.length;
+    const initialColumnCount = this._tiles[0].length;
+
+    if (rowsToPop > 0 || rowsToShift > 0) {
+      this._tiles = this._tiles.slice(
+        rowsToShift,
+        rowsToPop ? -rowsToPop : undefined
+      );
+    }
+    if (columnsToPop > 0 || columnsToShift > 0) {
+      this._tiles.forEach((row, rowIndex) => {
+        this._tiles[rowIndex] = this._tiles[rowIndex].slice(
+          columnsToShift,
+          columnsToPop ? -columnsToPop : undefined
+        );
+      });
+    }
+    // TODO: Instead of setting the dimensions directly, should it call a method on
+    // EditableTileMap that will iterates over all the layers to change their dimensions?
+    this.tileMap.setDimensionX(
+      initialColumnCount - columnsToPop - columnsToShift
+    );
+    this.tileMap.setDimensionY(initialRowCount - rowsToPop - rowsToShift);
+  }
+
+  increaseDimensions(
+    columnsToAppend: number,
+    columnsToUnshift: number,
+    rowsToAppend: number,
+    rowsToUnshift: number
+  ) {
+    const initialRowCount = this._tiles.length;
+    const initialColumnCount = this._tiles[0].length;
+    if (columnsToAppend > 0 || columnsToUnshift > 0) {
+      this._tiles.forEach((row, rowIndex) => {
+        const newRow = new Int32Array(
+          initialColumnCount + columnsToAppend + columnsToUnshift
+        ).fill(0);
+        newRow.set(row, columnsToUnshift);
+        this._tiles[rowIndex] = newRow;
+      });
+    }
+    if (rowsToAppend > 0 || rowsToUnshift > 0) {
+      this._tiles.unshift(
+        ...new Array(rowsToUnshift)
+          .fill(0)
+          .map(() =>
+            new Int32Array(
+              initialColumnCount + columnsToAppend + columnsToUnshift
+            ).fill(0)
+          )
+      );
```

Comment:
We should add a todo about over-provisionning to avoid to duplicate the map in loop when something is drawn toward the outside.